.chloggen/tp-error-handling.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
```

Comment:
I would personally err on the side of this being a breaking change, even if it is unlikely anyone relies on specific error handling behavior. My primary concern is around setting attributes used later on in a telemetry pipeline, and what the ramifications are if there is an error in an important statement.